modules/extensions-support/src/main/java/org/mule/module/extension/internal/runtime/resolver/NestedProcessorValueResolver.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.module.extension.internal.runtime.resolver;
+
+import org.mule.api.MuleEvent;
+import org.mule.api.MuleException;
+import org.mule.api.NestedProcessor;
+import org.mule.api.processor.MessageProcessor;
+import org.mule.processor.chain.NestedProcessorChain;
+
+/**
+ * A {@link ValueResolver} which wraps the given
+ * {@link MuleEvent} in a {@link NestedProcessor}.
+ * This resolver returns new instances per every invocation
+ *
+ * @since 3.7.0
+ */
+public final class NestedProcessorValueResolver implements ValueResolver<NestedProcessor>
```

Comment:
it's needed to pass a NestedProcessor as an argument to a method annotated with @Operation using the same mechanism that would use for a regular type of argument